luma/lcd/device.py

```diff
@@ -378,6 +378,138 @@ def command(self, cmd, *args):
+        self.command(0xed, 0x64, 0x03, 0x12, 0x81)       # Power on sequence control
+        self.command(0xe8, 0x85, 0x00, 0x78)             # Driver timing control A
+        self.command(0xcb, 0x39, 0x2c, 0x00, 0x34, 0x02) # Power control A
+        self.command(0xf7, 0x20)                         # Pump ratio control
+        self.command(0xea, 0x00, 0x00)                   # Driver timing control B
+        self.command(0xc0, 0x23)                         # Power Control 1, VRH[5:0]
+        self.command(0xc1, 0x10)                         # Power Control 2, SAP[2:0], BT[3:0]
+        self.command(0xc5, 0x3e, 0x28)                   # VCM Control 1
+        self.command(0xc7, 0x86)                         # VCM Control 2
+        self.command(0x36, 0x20 | order)                 # Memory Access Control
+        self.command(0x3a, 0x46)                         # Pixel Format 6-6-6
+        self.command(0xb1, 0x00, 0x18)                   # FRMCTR1
+        self.command(0xb6, 0x08, 0x82, 0x27)             # Display Function Control
+        self.command(0xf2, 0x00)                         # 3Gamma Function Disable
+        self.command(0x26, 0x01)                         # Gamma Curve Selected
+        self.command(0xe0,                               # Set Gamma (+ polarity)
+                     0x0f, 0x31, 0x2b, 0x0c, 0x0e, 0x08, 0x4e, 0xf1,
+                     0x37, 0x07, 0x10, 0x03, 0x0e, 0x09, 0x00)
+        self.command(0xe1,                               # Set Gamma (- polarity)
+                     0x00, 0x0e, 0x14, 0x03, 0x11, 0x07, 0x31, 0xc1,
+                     0x48, 0x08, 0x0f, 0x0c, 0x31, 0x36, 0x0f)
+        self.command(0x11)                               # Sleep out
+        self.clear()                               
+        self.show()
+
+    def display(self, image):
+        """
+        Renders a 24-bit RGB image to the ILI9341 LCD display. The 8-bit RGB
+        values are passed directly to the devices internal storage, but only
+        the 6 most-significant bits are used by the display.
+
+        :param image: The image to render.
+        :type image: PIL.Image.Image
+        """
+        assert(image.mode == self.mode)
+        assert(image.size == self.size)
+
+        image = self.preprocess(image)
+
+        if self.framebuffer.redraw_required(image):
+            left, top, right, bottom = self.apply_offsets(self.framebuffer.bounding_box)
+            width = right - left
+            height = bottom - top
+
+            self.command(0x2A, left >> 8, left & 0xFF, (right - 1) >> 8, (right - 1) & 0xFF)     # Set column addr
+            self.command(0x2B, top >> 8, top & 0xFF, (bottom - 1) >> 8, (bottom - 1) & 0xFF)     # Set row addr
+            self.command(0x2C)                                                                   # Memory write
+
+            self.data(self.framebuffer.image.crop(self.framebuffer.bounding_box).tobytes())
```

Comment:
Woah, this is remarkably simple 👍

Reply:
I don't love how it's going into implementation details of the framebuffer but I'm not sure if it's worth changing the framebuffer to support this in a more public way :/